styles/Canonical/021b-latinisms-to-reconsider.yml

```
@@ -0,0 +1,19 @@
+# 021 - Latinisms that we strongly advise against
+# Since Vale substitutions can't ignore word boundaries, meaning they can't detect the punctuation in "i.e." and "&c", we merely check for them and then notify the user.
+
+extends: existence
+message: |
+  Instead of Latin abbreviations, consider using an English equivalent.
+  Try not to use '&c', 'e.g.', 'i.e.', or 'n.b.'.
+  For 'a.m.', 'AM', 'p.m.', and 'PM', use 24-hour time.
+link: https://docs.ubuntu.com/styleguide/en#latin-words-and-phrases
+level: suggestion
+scope: raw
+nonword: true
+tokens:
+  - '&c'
+  - 'e\.g\.'
+  - 'i\.e\.'
+  - 'n\.b\.'
+  - '(a\.m\.|[A]\.?[M]\.?)'
+  - '(p\.m\.|[P]\.?[M]\.?)'
```

Be careful with raw and nonword.

Please add the \b to the terms without dots (you added because of this, right?)

For the others adding a leading \b would be more careful

Please note nonword:false is about surrounding each term with a \b

```suggestion
scope: raw
nonword: true
tokens:
  - '\b&c\b'
  - '\be\.g\.'
  - '\bi\.e\.'
  - '\bn\.b\.'
  - '\b(a\.m\.|[A]\.?[M]\.?)'
  - '\b(p\.m\.|[P]\.?[M]\.?)'
```

Otherwise, something like `whatever&cwhatever`would be caught. (Pain&Cie for example)

Updating the test to include such examples would be a good idea.

Also please check if you need scope: raw,
it could mess with URL, and other things

These comments are also valid for other rule files

---

Another example of raw+nonword would be  the reporting of any HTML document containing `&copy;`

---

Fixed in 3e52fdb10ef5759311fcdf5e49371afd4f8ea9a6